metadata-ingestion/src/datahub/emitter/mce_builder.py

```diff
@@ -69,7 +70,9 @@ def make_data_platform_urn(platform: str) -> str:
 def make_dataset_urn(platform: str, name: str, env: str = DEFAULT_ENV) -> str:
     if DATASET_URN_TO_LOWER:
         name = name.lower()
-    return f"urn:li:dataset:({make_data_platform_urn(platform)},{name},{env})"
+    dataset_urn = f"urn:li:dataset:({make_data_platform_urn(platform)},{name},{env})"
```

Comment:
Why don't you use the DatasetUrn here, I think that would do the validation automatically?
`DatasetUrn.create_from_ids(platform_id=platform, table_name=name, env=env)`

I think eventually we should ditch `make_dataset_urn` method and use `DatasetUrn` everywhere.